corehq/apps/cleanup/tasks.py

```diff
@@ -141,3 +146,47 @@ def pprint_stats(stats, outstream):
 @periodic_task(serializer='pickle', run_every=crontab(minute=0, hour=0), queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
 def clear_expired_sessions():
     call_command('clearsessions')
+
+
+@periodic_task(run_every=crontab(minute=0, hour=0), queue=getattr(settings, 'CELERY_PERIODIC_QUEUE', 'celery'))
+def check_for_sql_cases_without_existing_domain():
+    existing_domain_names = set(Domain.get_all_names())
+
+    case_domain_names = set()
+    db_names = get_db_aliases_for_partitioned_query()
+    for db_name in db_names:
+        case_domain_names |= set(CommCareCaseSQL.objects.using(db_name).values_list(
+            'domain', flat=True).distinct())
+
+    missing_domains_with_cases = case_domain_names - existing_domain_names
```

Comment:
couldn't you achieve the same thing by excluding cases with domains in the list?
```CommCareCaseSQL.objects.using(db_name).exclude(domain__in=existing_domain_names)```

Reply:
that's true - updated.